runtime/near-test-contracts/test-contract-rs/src/lib.rs

```diff
@@ -477,7 +477,8 @@ pub unsafe fn sum_n() {
 
     let mut sum = 0u64;
     for i in 0..n {
-        sum += i;
+        // NOTE: just sum += i optimizes into O(1) computation.
```

Comment:
Another option is to do something like naive prime factorization and adjust results per known complexity.